### ISSUE TITLE
Rewrite GPG check

### ISSUE THREAD

**Issue by jedfrechette** [wget/chocolatey_openvpn#7]
On a new machine with a fresh Chocolatey install attempting to install openvpn 2.4.6.20180710 fails due to a failure to import the PGP key. Below is an excerpt of the relevant lines from chocolatey.log:

```
2018-11-20 18:21:22,414 10724 [DEBUG] - Importing PGP key 'C:\ProgramData\chocolatey\lib\openvpn\tools\openvpn_public_key_new.asc' in the temporary keyring (C:\ProgramData\chocolatey\lib\openvpn\tools\591b7b5d-bccb-4a33-bddd-195a756dc932\pubring.gpg)...
2018-11-20 18:21:22,477 10724 [INFO ] - VERBOSE: gpg: keyblock resource '/c/Users/administrator/C:\ProgramData\chocolatey\lib\openvpn\tools\591b7b5d-bccb-4a33-bddd-195a756dc932/pubring.kbx': No such file or directory
2018-11-20 18:21:22,478 10724 [INFO ] - VERBOSE: gpg: pub  rsa4096/12F5F7B42F2B01E7 2017-02-09  OpenVPN - Security Mailing List <security@openvpn.net>
2018-11-20 18:21:22,479 10724 [INFO ] - VERBOSE: gpg: key 12F5F7B42F2B01E7: 22 signatures not checked due to missing keys
2018-11-20 18:21:22,479 10724 [INFO ] - VERBOSE: gpg: no writable keyring found: Not found
2018-11-20 18:21:22,480 10724 [INFO ] - VERBOSE: gpg: error reading 'C:\ProgramData\chocolatey\lib\openvpn\tools\openvpn_public_key_new.asc': General error
2018-11-20 18:21:22,481 10724 [INFO ] - VERBOSE: gpg: import from 'C:\ProgramData\chocolatey\lib\openvpn\tools\openvpn_public_key_new.asc' failed: General error
2018-11-20 18:21:22,481 10724 [INFO ] - VERBOSE: gpg: Total number processed: 0
2018-11-20 18:21:22,485 10724 [ERROR] - ERROR: Unable to import PGP key 'C:\ProgramData\chocolatey\lib\openvpn\tools\openvpn_public_key_new.asc' in the temporary keyring (C:\ProgramData\chocolatey\lib\openvpn\tools\591b7b5d-bccb-4a33-bddd-195a756dc932\pubring.gpg).
```

**Comment by jedfrechette** [comment]
After further investigation it appears that this error was occurring because the gpg.exe binary that was being found on the path was not the one shipped with gpg4win-vanilla. Instead a newer version that had been shipped with Git was detected and used.

**Comment by wget** [comment]
@jedfrechette thanks for the issue. Actually I plan to upstream GPG check directly with a future chocolatey version. The implementation will be based on the BountyCastle C# library, because it appears quite rock solid (and I didn't know they had a gpg implementation). I wanted to use this gpgme implementation at first, but may drop the idea: https://github.com/wget/gpgme-sharp

**Comment by sdktr** [comment]
Ran into this today. What did you do to work around it @jedfrechette ?

```
2019-02-18 10:11:48,465 20868 [DEBUG] - Importing PGP key 'C:\ProgramData\chocolatey\lib\openvpn\tools\openvpn_public_key_new.asc' in the temporary keyring (C:\ProgramData\chocolatey\lib\openvpn\tools\cfba9cc7-88a9-4a19-9515-96c6bec99bb2\pubring.gpg)...
2019-02-18 10:11:48,545 20868 [INFO ] - VERBOSE: gpg: keyblock resource '/c/C:\ProgramData\chocolatey\lib\openvpn\tools\cfba9cc7-88a9-4a19-9515-96c6bec99bb2/pubring.kbx': No such file or directory
2019-02-18 10:11:48,546 20868 [INFO ] - VERBOSE: gpg: pub  rsa4096/12F5F7B42F2B01E7 2017-02-09  OpenVPN - Security Mailing List <security@openvpn.net>
2019-02-18 10:11:48,548 20868 [INFO ] - VERBOSE: gpg: key 12F5F7B42F2B01E7: 22 signatures not checked due to missing keys
2019-02-18 10:11:48,550 20868 [INFO ] - VERBOSE: gpg: no writable keyring found: Not found
2019-02-18 10:11:48,552 20868 [INFO ] - VERBOSE: gpg: error reading 'C:\ProgramData\chocolatey\lib\openvpn\tools\openvpn_public_key_new.asc': General error
2019-02-18 10:11:48,554 20868 [INFO ] - VERBOSE: gpg: import from 'C:\ProgramData\chocolatey\lib\openvpn\tools\openvpn_public_key_new.asc' failed: General error
2019-02-18 10:11:48,555 20868 [INFO ] - VERBOSE: gpg: Total number processed: 0
2019-02-18 10:11:48,562 20868 [ERROR] - ERROR: Unable to import PGP key 'C:\ProgramData\chocolatey\lib\openvpn\tools\openvpn_public_key_new.asc' in the temporary keyring (C:\ProgramData\chocolatey\lib\openvpn\tools\cfba9cc7-88a9-4a19-9515-96c6bec99bb2\pubring.gpg).

**Comment by jedfrechette** [comment]
If I recall correctly, I ran in to this issue because I was working on a computer where Git had been installed with the option to add its binaries to the PATH selected. That put the version of gpg.exe shipped with git ahead of the one from this package on my PATH. In my case the Git binaries didn't actually need to be on the PATH so I just removed them from it, which allowed openvpn to find its version of gpg.exe.
